src/rust/engine/fs/fs_util/src/main.rs

```diff
@@ -410,19 +480,10 @@ pub fn fingerprint_to_digest(
   digest
 }
 
-fn file_name_as_utf8(path: &Path) -> Result<String, String> {
-  match path.file_name() {
-    Some(name) => {
-      match name.to_str() {
-        Some(name_utf8) => Ok(name_utf8.to_string()),
-        None => Err(format!(
-          "{:?}'s file_name is not representable in UTF8",
-          path
-        )),
-      }
-    }
-    None => Err(format!("{:?} did not have a file_name", path)),
-  }
+fn osstring_as_utf8(path: &OsStr) -> Result<String, String> {
```

Comment:
Given that this is going to be called using only single components, it could be hard to discover what the complete path was. Two options I see:
1. pass in the parent directory of the path component purely for use in the error message
2. `map_err` in the caller to contextualize which directory you found the bad component in

Reply:
Awkwardly, we've thrown away the directory information at this point... We still have the Stat underlying the PathStat, but that may be a symlink target... I'm vaguely tempted to just leave this as a not-ideal edge case for now, because it should basically never happen, and running `find` shouldn't be too big a deal if it does. Having to pass the context of the calls which recursed into us around feels icky.